### PR TITLE
[Avalon-6] Bumps bootstrap-sass from 3.3.3 to 3.4.1.

### PR DESCRIPTION
**Pull request by jefferya**
Addresses: 
- #440 
- https://blog.getbootstrap.com/2019/02/13/bootstrap-4-3-1-and-3-4-1/

Bumps [bootstrap-sass](https://github.com/twbs/bootstrap-sass) from 3.3.3 to 3.4.1. 
<details>
<summary>Vulnerabilities fixed</summary>

*Sourced from The GitHub Security Advisory Database.*

> **Moderate severity vulnerability that affects bootstrap and bootstrap-sass**
> In Bootstrap 4 before 4.3.1 and Bootstrap 3 before 3.4.1, XSS is possible in the tooltip or popover data-template attribute. For more information, see: https://blog.getbootstrap.com/2019/02/13/bootstrap-4-3-1-and-3-4-1/
> 
> Affected versions: >= 3.0.0, < 3.4.1

</details>

### CODE REVIEW

File: app/assets/javascripts/button_confirmation.js.coffee [hunk 1]
```diff
@@ -35,9 +36,16 @@ $ ->
...
+    originalMethodWhiteList = dataMethodWhiteList.a.slice()
+    dataMethodWhiteList.a.push('data-method')
     t = this
     $('.btn-confirmation').filter(->
       this != t
     ).popover 'hide'
     $(this).popover 'show'
+    # return Bootstrap whitelist to original
+    dataMethodWhiteList.a = originalMethodWhiteList
     false
```
Comment (jefferya) [inline app/assets/javascripts/button_confirmation.js.coffee]: Not certain the best way to relax the new sanitation requirements in bootstrap-sass 3.4.1. I've never touched coffeescript before this task so open for suggestions.

The above approach adds the `data-method` then removes from the whitelist. References 
- https://blog.getbootstrap.com/2019/02/13/bootstrap-4-3-1-and-3-4-1/
- https://getbootstrap.com/docs/3.4/javascript/#js-sanitizer

Another option, decided against disabling sanitation for the popup:  
https://github.com/ualbertalib/avalon/commit/eca7f963e0e85e0ac0dbbf361124e3116879583d

File: app/assets/javascripts/button_confirmation.js.coffee [hunk 2]
```diff
@@ -17,6 +17,7 @@
 # to all disjointed submit elements as is needed in some of the workflow steps
 
 $ ->
+  dataMetList = $.fn.tooltip.Constructor.DEFAULTS.whiteList
```
Comment (pgwillia) [inline app/assets/javascripts/button_confirmation.js.coffee]: I'm probably missing something obvious... Where is this used?

File: app/assets/javascripts/button_confirmation.js.coffee [hunk 3]
```diff
@@ -35,9 +36,16 @@ $ ->
         $('#' + $(this).attr('form')).find('[name=\'_method\']').val 'delete'
       '<p>Are you sure?</p> ' + button + ' <a href="#" class="btn btn-xs btn-primary" id="special_button_color">No, Cancel</a>'
   ).click ->
+    # Bootstrap whitelist: add 'data-method as per bootstrap 3.4.1 update with new sanitizer code
+    # https://getbootstrap.com/docs/3.4/javascript/#js-sanitizer
+    dataMethodWhiteList = $.fn.tooltip.Constructor.DEFAULTS.whiteList
+    originalMethodWhiteList = dataMethodWhiteList.a.slice()
+    dataMethodWhiteList.a.push('data-method')
```
Comment (pgwillia) [inline app/assets/javascripts/button_confirmation.js.coffee]: This must be about line 32/33 `data-method="delete"`

File: app/assets/javascripts/button_confirmation.js.coffee [hunk 4]
```diff
@@ -35,9 +36,16 @@ $ ->
...
     t = this
     $('.btn-confirmation').filter(->
       this != t
     ).popover 'hide'
     $(this).popover 'show'
```
Comment (pgwillia) [inline app/assets/javascripts/button_confirmation.js.coffee]: So this must be where the new sanitizer would remove the attributes not on the whitelist.

File: app/assets/javascripts/button_confirmation.js.coffee [hunk 5]
```diff
@@ -35,9 +36,16 @@ $ ->
...
+    # return Bootstrap whitelist to original
+    dataMethodWhiteList.a = originalMethodWhiteList
```
Comment (pgwillia) [inline app/assets/javascripts/button_confirmation.js.coffee]: And then you're returning to the original whitelist. Do you need to do this?

File: app/assets/javascripts/button_confirmation.js.coffee (hunk 3 above)
Comment (jefferya) [reply app/assets/javascripts/button_confirmation.js.coffee]: Correct.

> This must be about line 32/33 `data-method="delete"`

Correct.

File: app/assets/javascripts/button_confirmation.js.coffee (hunk 5 above)
Comment (jefferya) [reply app/assets/javascripts/button_confirmation.js.coffee]: > And then you're returning to the original whitelist. Do you need to do this?

Unsure. Though I'd be on the safe side. Not certain of scope, especially within coffeescript.

File: app/assets/javascripts/button_confirmation.js.coffee (hunk 2 above)
Comment (pgwillia) [reply app/assets/javascripts/button_confirmation.js.coffee]: This was an old commit?  Sorry about that.

**Pull request merged**